### ISSUE TITLE
📝 add documentation page for client implementation

### ISSUE THREAD

**Issue by echavarria-mqv** [Munich-Quantum-Software-Stack/QDMI#106]
I think we should consider adding the example of a `QDMI client`.

**Comment by burgholzer** [comment]
Examples of a FoMaC and a tool (=client) are already part of the examples and in the documentation.

**Comment by echavarria-mqv** [comment]
Why is it called tool instead of client though?

**Comment by burgholzer** [comment]
Because it is an example of a tool using QDMI (specifically, the client interface) and the example FoMaC.

**Comment by echavarria-mqv** [comment]
If it is effectively the example of a client, I suggest to rename it as client to keep it consistent.

**Comment by burgholzer** [comment]
I see your point. However, I'd still have a slight preference of keeping the naming as is.
The FoMaC library also uses the client interface (and, thus is a client per that definition) and it still has it's own name.
Both the FoMaC and the Tool are example QDMI clients.

@ystade any opinion on that?

**Comment by ystade** [comment]
I think the important point is that we should add a client page to the example page in the documentation. Regarding the naming I do not have strong preference, I would just leave it as is.

**Comment by burgholzer** [comment]
> I think the important point is that we should add a client page to the example page in the documentation.

Re-opening this so that it can be closed once said page has been created.

**Comment by echavarria-mqv** [comment]
I've been asked more than once for an example of a QDMI client. Which is why I opened this issue. I propose to move the FoMaC example to the FoMaC repo and rename the tool example in this repo to client.

**Comment by ystade** [comment]
This has the problem that the tool cannot work without the fomac. I would prefer if the QDMI repository is self contained.

**Comment by burgholzer** [comment]
> I've been asked more than once for an example of a QDMI client. Which is why I opened this issue. I propose to move the FoMaC example to the FoMaC repo and rename the tool example in this repo to client.

The same argument could be made for the example backends in this repository.
If we were to remove all of these, then the QDMI library would simply contain the header-only library for the interface and nothing else.
That would not be very helpful, as examples would be spread-out all across the repositories and it would be extremely hard to test the QDMI repository.
All the examples being here allows us to properly test the interface, to document it, and to provide example implementations for all parts of it.

I agree that the documentation could be slightly polished.

**Comment by echavarria-mqv** [comment]
No, I think that a minimally working QDMI client and a minimally working QDMI device should be part of this repo. We'll also need to provide examples for the other repositories as well (e.g., `backends`, `passes`, `QRM`, `FoMaC`, etc.) anyway.
